microsoft/testsuites/core/provisioning.py

```diff
@@ -149,7 +149,57 @@ def verify_deployment_provision_sriov(
     ) -> None:
         self._smoke_test(log, node, log_path)
 
-    def _smoke_test(self, log: Logger, node: RemoteNode, log_path: Path) -> None:
+    @TestCaseMetadata(
+        description="""
+        This case runs smoke test on a node provisioned.
+        The test steps are almost the same as `smoke_test` except for
+        executing reboot from Azure SDK.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+            network_interface=Sriov,
+            supported_features=[SerialConsole],
+        ),
+    )
+    def verify_reboot_from_api(
+        self, log: Logger, node: RemoteNode, log_path: Path
+    ) -> None:
+        self._smoke_test(log, node, log_path, reboot_from_shell=False)
+
+    @TestCaseMetadata(
+        description="""
+        This case runs smoke test on a node provisioned.
+        The test steps are almost the same as `smoke_test` except for
+        executing stop then start from Azure SDK.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+            network_interface=Sriov,
+            supported_features=[SerialConsole],
+        ),
+    )
+    def verify_stop_start_from_api(
+        self, log: Logger, node: RemoteNode, log_path: Path
+    ) -> None:
+        self._smoke_test(
+            log,
+            node,
+            log_path,
+            reboot_from_shell=False,
+            deallocated=True,
+        )
+
+    def _smoke_test(
```

Comment:
This is the right place to merge feature/tool restart together, not in the reboot tool.